src/Core/Settings.h

```diff
@@ -155,6 +155,8 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     \
     M(Bool, allow_experimental_parallel_reading_from_replicas, false, "If true, ClickHouse will send a SELECT query to all replicas of a table. It will work for any kind on MergeTree table.", 0) \
     \
+    M(UInt64, system_replicas_fetch_threads, 16, "The maximum number of threads to fetch data for system.replicas table.", 0) \
```

Comment:
Do we really need a setting for this?

Reply:
We could take the number of threads from the `max_treads` (and we do), but I am not sure which value should be set in the case when `max_threads = 0`. Any suggestions would be appreciated :)

Reply:
max_threads = 0 usually means "use all physical cores"